### ISSUE TITLE
lock_api: Bug in example spinlock implementation

### ISSUE THREAD

**Issue by phil-opp** [Amanieu/parking_lot#215]
The example spinlock implementation in https://docs.rs/lock_api/0.3.3/lock_api/#example does not seem to be correct. The problem is the `try_lock` method:

```rust
fn try_lock(&self) -> bool {
    self.0.swap(true, Ordering::Acquire)
}
```

The documentation isn't explicit about this, but I assume that the function should return `true` if the lock was successfully acquired. The problem is that `swap` returns the previous value, so this function returns true only if the lock is already acquired. So I think it should be `!self.0.swap(true, Ordering::Acquire)` instead.

**Comment by Amanieu** [comment]
This has already been fixed in master, see #207.

**Comment by phil-opp** [comment]
Ah, I did not notice this, sorry the duplicate report! Maybe there should be a new crates.io release for this, given that it recommends a broken spinlock implementation?
